### PR TITLE
Refactor - Retrospective Assessment

### PR DESCRIPTION
**Pull request by joefitter**
* Added new RA component
* if conditions are met, it is always true and uneditable
* if not then try and get value from legacy

### CODE REVIEW

File: client/components/conditions/index.js [hunk 1]
```diff
@@ -5,6 +5,7 @@ import { Button } from '@ukhomeoffice/react-components';
 import { updateConditions } from '../../actions/projects';
 import Conditions from './conditions';
 import AddConditions from './add-conditions';
+import RA from '../retrospective-assessment';
```
Comment (wheelsandcogs) [inline client/components/conditions/index.js]: Do we really need to abbreviate this?

File: client/pages/sections/granted/project-summary.js [hunk 2]
```diff
@@ -1,10 +1,9 @@
 import React, { Fragment } from 'react';
-import last from 'lodash/last';
 import { connect } from 'react-redux';
 import ReactMarkdown from 'react-markdown';
 import Review from '../../../components/review';
+import RA from '../../../components/retrospective-assessment';
```
Comment (wheelsandcogs) [inline client/pages/sections/granted/project-summary.js]: Do we really need to abbreviate this?

File: client/pages/sections/legacy-conditions.js [hunk 3]
```diff
@@ -1,18 +1,15 @@
 import React, { Fragment } from 'react';
 import { connect } from 'react-redux';
 import ReactMarkdown from 'react-markdown';
-import { updateConditions, updateRetrospectiveAssessment } from '../../actions/projects';
+import { updateConditions } from '../../actions/projects';
 import AddConditions from '../../components/conditions/add-conditions';
 import Condition from '../../components/conditions/condition';
-import Fieldset from '../../components/fieldset';
-import ReviewFields from '../../components/review-fields';
+import RA from '../../components/retrospective-assessment';
```
Comment (wheelsandcogs) [inline client/pages/sections/legacy-conditions.js]: same as above...

File: views/pdf/v0/index.jsx [hunk 4]
```diff
@@ -4,7 +4,7 @@ import { connect } from 'react-redux';
 import ReactMarkdown from 'react-markdown';
 import { getSubsections } from '../../../client/schema';
 import StaticSection from '../../../client/components/static-section';
-import ReviewFields from '../../../client/components/review-fields';
+import RA from '../../../client/components/retrospective-assessment';
```
Comment (wheelsandcogs) [inline views/pdf/v0/index.jsx]: see above.

File: client/components/retrospective-assessment.js [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+  if (!isUndefined(project.retrospectiveAssessment)) {
+    // legacy licences may have an object containing a boolean/
+    if (isPlainObject(project.retrospectiveAssessment)) {
+      return project.retrospectiveAssessment['retrospective-assessment-required'];
+    }
+    // now saved as a boolean
+    return !!project.retrospectiveAssessment;
+  }
+  // previous new licences contained a 'retrospective-assessment' condition.
+  if (project.conditions && project.conditions.find(c => c.key.match(/^retrospective-assessment$/))) {
```
Comment (lennym) [inline client/components/retrospective-assessment.js]:
```suggestion
  if (project.conditions && project.conditions.find(c => c.key === 'retrospective-assessment')) {
```

**Pull request merged**